src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int FIRST_INDEX = 1;
+    private static final int SECOND_INDEX = 2;
+    private static final int THIRD_INDEX = 3;
+    private static final int AT_LEAST_ONE_ROW = 1;
+    private static final int ZERO_ROWS = 0;
+    private static final String DELETE_BY_ID_REQUEST = "DELETE FROM books WHERE id = ?";
+    private static final String UPDATE_REQUEST
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String FIND_ALL_REQUEST = "SELECT * FROM books";
+    private static final String FIND_BY_ID_REQUEST = "SELECT * FROM books WHERE id = ?";
+    private static final String CREATION_REQUEST = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String TITLE = "title";
+    private static final String ID = "id";
+    private static final String PRICE = "price";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(CREATION_REQUEST, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(FIRST_INDEX, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX, book.getPrice());
+            if (statement.executeUpdate() < AT_LEAST_ONE_ROW) {
+                throw new RuntimeException("Excepted "
+                        + "to insert at least one row, but inserted 0 rows.");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_REQUEST)) {
+            statement.setLong(FIRST_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(resultSet.getString(TITLE));
+                book.setPrice(resultSet.getObject(PRICE, BigDecimal.class));

```

Comment:
Here and in `findAll` consider moving mapping from result set to the model to separate method